python/lsst/daf/butler/registry/tests/_registry.py

```diff
@@ -127,6 +126,24 @@ def loadData(self, registry: Registry, filename: str):
         backend.register()
         backend.load(datastore=None)
 
+    def checkQueryResults(self, results, expected):
+        """Check that a query results object contains expected values.
+
+        Parameters
+        ----------
+        results : `DataCoordinateQueryResults` or `DatasetQueryResults`
+            A lazy-evaluation query results object.
+        expected : `list`
+            A list of `DataCoordinate` o `DatasetRef` objects that should be
+            equal to results of the query, aside from ordering.
+        """
+        self.assertCountEqual(list(results), expected)
+        self.assertEqual(results.count(), len(expected))
+        if expected:
+            self.assertTrue(results.any())
+        else:
+            self.assertFalse(results.any())
```

Comment:
Is not `any()` allowed to be true even if `expected` is empty. Or do we only test cases when `any` is definitely false, or expected is never empty?